sky/packages/sky/lib/src/gestures/arena.dart

```diff
@@ -129,20 +137,32 @@ class GestureArena {
     _GestureArenaState state = _arenas[key];
     if (state == null)
       return;  // This arena has already resolved.
-    assert(!state.isOpen);
     assert(state.members.contains(member));
     if (disposition == GestureDisposition.rejected) {
       state.members.remove(member);
       member.rejectGesture(key);
-      _tryToResolveArena(key, state);
+      if (!state.isOpen)
+        _tryToResolveArena(key, state);
     } else {
       assert(disposition == GestureDisposition.accepted);
-      _arenas.remove(key);
-      for (GestureArenaMember rejectedMember in state.members) {
-        if (rejectedMember != member)
-          rejectedMember.rejectGesture(key);
+      if (state.isOpen) {
+        state.eagerWinner = member;
```

Comment:
What if there's already an eagerWinner?  Usually we have the first one in the event path win, which would mean whoever gets into the eagerWinner slot first.